server/index.js

```diff
@@ -56,6 +56,110 @@ app.get('/api/products/:productId', (req, res, next) => {
   }
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const sqlEmpty = `
+  SELECT *
+  FROM "carts"
+  `;
+
+  const sqlExist = `
+  SELECT "cartItems"."cartItemId",
+          "cartItems"."price",
+          "products"."productId",
+          "products"."image",
+          "products"."name",
+          "products"."shortDescription"
+  FROM "cartItems"
+  JOIN "products" USING ("productId")
+  WHERE "cartItems"."cartId" = $1
+  `;
+
+  db.query(!req.session.cartId ? sqlEmpty : sqlExist, [req.session.cartId])
+    .then(result => {
+      const cartContent = result.rows;
+      res.status(200).json(cartContent);
+    });
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId;
+  const sql = `
+  SELECT "price"
+  FROM "products"
+  WHERE "productId" = $1
+  `;
+  if (parseInt(productId) < 0 || !productId) {
+    next(new ClientError(`${productId} is not a Valid Product ID`, 400));
+  } else {
+    db.query(sql, [productId])
+      .then(result => {
+        const productIdCheck = result.rows;
+        if (!productIdCheck) {
+          throw new ClientError('That is an invalid request');
+        } else {
+          const insertNewCartSQL = `
+              INSERT INTO "carts" ("cartId", "createdAt")
+              VALUES (default, default)
+              RETURNING "cartId"
+            `;
+
+          const insertExistingCartSQL = `
+              SELECT "cartId"
+              FROM "carts"
+              WHERE "cartId" = $1
+          `;
```

Comment:
If you already have the `cartId`, you don't need to query the database for it. Just use an `if/else` that may return an object with the `cartId` and `price`.